rust/lance/src/dataset/fragment.rs

```diff
@@ -641,6 +641,18 @@ impl FileFragment {
             }
         }
 
+        if self.metadata.files.iter().any(|f| f.is_legacy_file())
+            != self.metadata.files.iter().all(|f| f.is_legacy_file())
+        {
+            return Err(Error::corrupt_file(
+                self.dataset
+                    .data_dir()
+                    .child(self.metadata.files[0].path.as_str()),
+                "Fragment contains a mix of v1 and v2 data files".to_string(),
+                location!(),
+            ));
+        }
```

Comment:
So is the rule that different fragments can have different versions of files, but within the same fragment they all need to be the same version?